components/tools/OmeroJava/test/integration/LightAdminRolesTest.java

```diff
@@ -2437,4 +2453,173 @@ public void testModifyGroupEdit(boolean isPrivileged,
+    public void testDeleteEnumerationsbyRestrictedSystemUser() throws Exception {
+        //root create an enumeration first
+        ContrastMethod ho = new ContrastMethodI();
+        ho.setValue(omero.rtypes.rstring("testDeleteEnumerationsbyRestrictedSystemUser"));
+        final ITypesPrx ts = root.getSession().getTypesService();
+        List<IObject> types = ts.allEnumerations(ContrastMethod.class.getName());
+        int n = types.size();
+        ho = (ContrastMethod) ts.createEnumeration(ho);
+        types = ts.allEnumerations(ContrastMethod.class.getName());
+        Assert.assertEquals(types.size(), (n+1));
+        logNewAdminWithoutPrivileges();
+        //try to delete an enum type. The type is not important
+        final ITypesPrx types_svc = factory.getTypesService();
+        types = types_svc.allEnumerations(ContrastMethod.class.getName());
+        int m = types.size();
+        Assert.assertEquals(m, (n+1));
+        Iterator<IObject> i = types.iterator();
+        while (i.hasNext()) {
+            IObject o = i.next();
+            if (o.getId().getValue() == ho.getId().getValue()) {
+                types_svc.deleteEnumeration(ho);
+                break;
+            }
+        }
+        types = types_svc.allEnumerations(ContrastMethod.class.getName());
+        Assert.assertEquals(types.size(), n);
+    }
+
+    /**
+     * Tests if an enumeration already used can be deleted.
+     * This should fail.
+     * @throws Exception
+     */
+    @Test(expectedExceptions = omero.ValidationException.class)
+    public void testDeleteUsedEnumerationsbyRestrictedSystemUser() throws Exception {
+        logNewAdminWithoutPrivileges();
+        //try to delete an enum type. The type is not important
+        final ITypesPrx types_svc = factory.getTypesService();
+        List<IObject> types = types_svc.allEnumerations(EventType.class.getName());
+        Iterator<IObject> i = types.iterator();
+        while (i.hasNext()) {
+            types_svc.deleteEnumeration(i.next());
+        }
+    }
+
+    /**
+     * Tests if a deleted enumeration can be restored.
+     * @throws Exception
+     */
+    @Test
+    public void testResetEnumerationsbyRestrictedSystemUser() throws Exception {
+      //root create an enumeration first
+        ContrastMethod ho = new ContrastMethodI();
+        ho.setValue(omero.rtypes.rstring("testResetEnumerationsbyRestrictedSystemUser"));
+        final ITypesPrx ts = root.getSession().getTypesService();
+        List<IObject> types = ts.allEnumerations(ContrastMethod.class.getName());
+        int n = types.size();
+        ho = (ContrastMethod) ts.createEnumeration(ho);
+        types = ts.allEnumerations(ContrastMethod.class.getName());
+        Assert.assertEquals(types.size(), (n+1));
+        logNewAdminWithoutPrivileges();
+        final ITypesPrx types_svc = factory.getTypesService();
+        types = types_svc.allEnumerations(ContrastMethod.class.getName());
+        int m = types.size();
```

Comment:
not sure what the thinking is where where we use `m` sometimes and `n+1` othertimes

Reply:
n is the original size of the list before a new enum is added
m is the size of the list after adding an enum. that's why we have check m and n+1

Reply:
I am probably a bit confused too here, just like @mtbc . I read this test as a sequence:
   - root creates an enum, raising the number of enums from n to n+1
   - light admin double-checks the number of enums, and for them it is m
   - m is equal to n+1
   - light admin resets the enums (this is probably the restoring step)
   - the size of the enums is m after the restoration, which is not surprising

I am missing the [deletion step](https://github.com/openmicroscopy/openmicroscopy/pull/5502/files#diff-d91232ef1e245a87e311c4a79a2a5f77R2484) which would precede the restoration step ?

Reply:
The test has now been reviewed.